Sources/SwiftProtobuf/BinaryEncoder.swift

```diff
@@ -133,6 +136,23 @@ internal struct BinaryEncoder {
         }
     }
 
+    // Write a string field, including the leading index/tag value.
+    mutating func putStringValue(value: String) {
+        #if swift(>=5.0)
+            // If the String does not support an internal representation in a form
+            // of contiguous storage, body is not called and nil is returned.
+            let isAvailable = value.utf8.withContiguousStorageIfAvailable { (body: UnsafeBufferPointer<UInt8>) -> Int in
+                putVarInt(value: body.count)
+                return append(contentsOf: UnsafeRawBufferPointer(body))
+            }
+            if isAvailable == nil {
+                loopWrite(value: value)
+            }
+        #else
+            loopWrite(value: value)
+        #endif
+    }
+
```

Comment:
You can probably just do something like this to shrink the code and make the edit when our min lets us remove the conditional simpler:

```swift
    mutating func putStringValue(value: String) {
      #if swift(>=5.0)
        // If the String does not support an internal representation in a form
        // of contiguous storage, body is not called and nil is returned.
        let isAvailable = value.utf8.withContiguousStorageIfAvailable { (body: UnsafeBufferPointer<UInt8>) -> Int in
            putVarInt(value: body.count)
            return append(contentsOf: UnsafeRawBufferPointer(body))
        }
      #else
        let isAvailable: String? = nil 
      #endif
        if isAvailable == nil {
            let count = value.utf8.count
            putVarInt(value: count)
            for b in value.utf8 {
                pointer.storeBytes(of: b, as: UInt8.self)
                pointer = pointer.advanced(by: 1)
            }
        }
    }
```

Reply:
It's your code base. It is so in 3aa2905bd4221bf9fd7fd481920b1b37aace1362.